### PR TITLE
fix: update AWS Bedrock API call and enhance error handling in bedrock_client.py; improve test cases for better coverage

### PR DESCRIPTION
**Pull request by S-mishina**
This pull request introduces several updates to enhance the functionality of the HCL Processor, particularly around its integration with AWS Bedrock and its configuration handling. The changes include improving error handling, updating the system prompt and output schema for better clarity, and replacing deprecated methods with new ones for interacting with the Bedrock API. Additionally, the test suite has been updated to reflect these changes.

### Updates to AWS Bedrock Integration:

* **Error Handling Enhancements**: Improved the error message for missing AWS credentials to provide clearer guidance when environment variables are not set. (`src/hcl_processor/bedrock_client.py`)
* **Refactored Bedrock API Calls**: Replaced the deprecated `invoke_model` method with the new `converse` method. This includes restructuring the payload into `messages`, `system`, `inferenceConfig`, and `toolConfig` for better modularity and extensibility. (`src/hcl_processor/bedrock_client.py`)

### Configuration and Schema Updates:

* **System Prompt Update**: Revised the `system_prompt` in `sample/bedrock_config.yaml` to provide more detailed and structured instructions for analyzing Terraform code and generating JSON output. (`sample/bedrock_config.yaml`)
* **Output Schema Standardization**: Updated the output schema in `sample/bedrock_config.yaml` to use consistent English field names (e.g., `monitor_name`, `type`) instead of Japanese, aligning with standard naming conventions. (`sample/bedrock_config.yaml`) [[1]](diffhunk://#diff-bb2bb9b515d579ac5252da49d99994ec96160fc9271c653daf7d415d61c3974eL37-R56) [[2]](diffhunk://#diff-bb2bb9b515d579ac5252da49d99994ec96160fc9271c653daf7d415d61c3974eL72-R87)

### Testing Improvements:

* **Updated Mock Responses**: Adjusted test cases to reflect the new `converse` method and updated response structure. This includes verifying the inclusion of `toolConfig` in API calls and ensuring proper handling of tool use responses. (`tests/test_bedrock_client.py`) [[1]](diffhunk://#diff-b899208b72e893436770c1e6f8b8618c4cea9796a06ca7533725006e393268eeL53-R86) [[2]](diffhunk://#diff-b899208b72e893436770c1e6f8b8618c4cea9796a06ca7533725006e393268eeL84-R101)
* **Enhanced Error Case Coverage**: Refined test cases to handle scenarios like missing response messages, invalid JSON formats, and other edge cases. (`tests/test_bedrock_client.py`) [[1]](diffhunk://#diff-b899208b72e893436770c1e6f8b8618c4cea9796a06ca7533725006e393268eeL94-R112) [[2]](diffhunk://#diff-b899208b72e893436770c1e6f8b8618c4cea9796a06ca7533725006e393268eeL105-R130)

### Miscellaneous:

* **Version Bump**: Incremented the project version from `0.10.3-a` to `0.10.4-a` in `pyproject.toml`. (`pyproject.toml`)

### CODE REVIEW

File: src/hcl_processor/bedrock_client.py [hunk 1]
```diff
@@ -82,48 +82,92 @@ def aws_bedrock(prompt, modules_data, config, system_config):
...
-
-    payload = {
-        "system": final_system_prompt,
-        "messages": [{"role": "user", "content": prompt}],
-        "anthropic_version": config["bedrock"]["payload"].get(
-            "anthropic_version",
-            system_config["default_bedrock"]["payload"]["anthropic_version"],
-        ),
-        "max_tokens": config["bedrock"]["payload"].get(
+    logger.info(f"Final system prompt: {final_system_prompt}")
```
Comment (S-mishina) [inline src/hcl_processor/bedrock_client.py:89]:
```suggestion
```
